theseus/core/loss.py

```diff
@@ -0,0 +1,42 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import abc
+
+import torch
+
+
+class Loss:
+    @staticmethod
+    @abc.abstractmethod
+    def evaluate(x: torch.Tensor, radius: torch.Tensor) -> torch.Tensor:
+        pass
+
+    @staticmethod
+    @abc.abstractmethod
+    def linearize(x: torch.Tensor, raidus: torch.Tensor) -> torch.Tensor:
+        pass
+
+
+class WelschLoss(Loss):
+    @staticmethod
+    def evaluate(x: torch.Tensor, radius: torch.Tensor) -> torch.Tensor:
```

Comment:
Are there restrictions on the sizes of x and/or radius? Should we explicitly check for batch? If these are common checks then we can move it to the base class.